### PR TITLE
skip empty srcs for safe iframe srcs

### LINKED ISSUE #3215: unsafe_html fails on all iframes if BUILD_LIVE_SAMPLES_BASE_URL is empty
See https://github.com/mdn/content/pull/3064/checks?check_run_id=2095600358
```
Injecting flaws into files/en-us/mdn/kitchensink/index.html (/en-US/docs/MDN/Kitchensink) failed.
Error: unsafe_html flaws: <script> tag found,Unsafe <iframe> 'src' value (https://interactive-examples.mdn.mozilla.net/pages/tabbed/abbr.html),Unsafe <iframe> 'src' value (https://interactive-examples.mdn.mozilla.net/pages/css/order.html),Unsafe <iframe> 'src' value (https://interactive-examples.mdn.mozilla.net/pages/js/regexp-assertions.html),Unsafe <iframe> 'src' value (/en-US/docs/Games/Techniques/2D_collision_detection/_samples_/Rect_code),Unsafe <iframe> 'src' value (/en-US/docs/MDN/Kitchensink/_samples_/inheritance_diagram),Unsafe <iframe> 'src' value (https://mdn.github.io/web-tech-games/index.html),'onerror' on-handler found in h2
```

It's a bit hard and ugly to see but it's basically complaining about the `https://interactive-examples.mdn.mozilla.net/pages/tabbed/abbr.html` iframe `src`. 
It happens because...
```
export BUILD_LIVE_SAMPLES_BASE_URL=
```

so now the list of `safeIFrameSrcs` contains an entry which is an empty string.

### PR DESCRIPTION
**Pull request by peterbe**
Fixes #3215

It looks like this now:

Screen Shot 2021-03-12 at 10 43 39 AM

Contrast that with: https://github.com/mdn/content/pull/3064/checks?check_run_id=2095600358

### CODE REVIEW

**Comment by peterbe** [comment]
To test this, run:
```
export BUILD_LIVE_SAMPLES_BASE_URL=
BUILD_FLAW_LEVELS="unsafe_html: error, *:ignore"
```
and then:
```
yarn build [specificfiles ...]
```

File: build/flaws.js [hunk 1]
```diff
@@ -114,12 +118,22 @@ function injectUnsafeHTMLFlaws(doc, $, { rawContent }) {
...
+  if (INTERACTIVE_EXAMPLES_BASE_URL) {
+    safeIFrameSrcs.push(INTERACTIVE_EXAMPLES_BASE_URL.toLowerCase());
+  }
 
   $("script, embed, object, iframe").each((i, element) => {
     const { tagName } = element;
     if (tagName === "iframe") {
       // For iframes we only check the 'src' value
       const src = $(element).attr("src");
+      if (src.startsWith("/") && !src.includes("://")) {
```
Comment (escattone) [inline build/flaws.js:133]: Should we also check for `!src.startsWith("//")` as well?

Comment (peterbe) [reply build/flaws.js:133]: You're right!

**Review by escattone — APPROVED** [review]
Thanks @peterbe!

**Pull request merged**